core/src/main/java/com/google/cloud/sql/core/InternalConnectorRegistry.java

```diff
@@ -263,4 +290,39 @@ private Connector createConnector(ConnectorConfig config) {
         refreshTimeoutMs,
         serverProxyPort);
   }
+
+  /** Register the configuration for a named connector. */
+  public void register(String name, ConnectorConfig config) {
+    if (this.namedConnectors.containsKey(name)) {
+      throw new IllegalArgumentException("Named connection " + name + " exists.");
+    }
+    this.namedConnectors.put(name, createConnector(config));
+  }
+
+  /** Close a named connector, stopping the refresh process and removing it from the registry. */
+  public void close(String name) {
+    Connector connector = namedConnectors.remove(name);
+    if (connector == null) {
+      throw new IllegalArgumentException("Named connection " + name + " does not exist.");
+    }
+    connector.close();
+  }
+
+  /** Shutdown all connectors and remove the singleton instance. */
+  public void shutdown() {
+    this.unnamedConnectors.forEach(
+        (key, c) -> {
+          c.close();
+        });
+    this.unnamedConnectors.clear();
```

Comment:
We need to close all the unnamed connectors in the `connectors` map.

Reply:
Fixed.